bundles/org.openhab.binding.doorbird/src/main/java/org/openhab/binding/doorbird/internal/api/DoorbirdSession.java

```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright (c) 2010-2024 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.doorbird.internal.api;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.doorbird.internal.model.GetsessionDTO;
+import org.openhab.binding.doorbird.internal.model.GetsessionDTO.GetsessionBha;
+
+import com.google.gson.JsonSyntaxException;
+
+/**
+ * The {@link DoorbirdSession} holds information about the Doorbird session,
+ * including the v2 decryption key for notification events.
+ *
+ * @author Mark Hilbush - Initial contribution
+ */
+@NonNullByDefault
+public class DoorbirdSession {
+    private @Nullable String returnCode;
+    private @Nullable String sessionId;
+    private @Nullable String decryptionKey;
+
+    @SuppressWarnings("null")
```

Comment:
Would it be possible to prevent this null suppression?

Reply:
I'm not sure what to do about this. `fromJson` can return null, but Eclipse doesn't think so. So when i remove the null suppression, Eclipse complains about a redundant null check on line 37. I guess I could remove the null suppression, but it bothers me when Eclipse complains lol. I can go either way, so whatever you want to do is fine with me.

Reply:
Null annotations can be a pain. :-/ As `fromJson` is not properly annotated, this is an acceptable warning. I prefer to have a warning over a suppresion, as that may lead to future issues.

Reply:
Fair enough. I'll remove the null suppression.